cartoframes/context.py

```diff
@@ -448,24 +432,36 @@ def fetch(self, query, decode_geom=False):
                 )
 
         """
-        copy_query = 'COPY ({query}) TO stdout WITH (FORMAT csv, HEADER true)'.format(query=query)
-        result = recursive_read(self, copy_query)
-
-        query_columns = get_columns(self, query)
-        df_types = dtypes(query_columns, exclude_dates=True, exclude_the_geom=decode_geom)
-        date_column_names = date_columns_names(query_columns)
+        with tqdm(desc='Executing query, please wait...', unit='rows') as t:
+            copy_query = 'COPY ({query}) TO stdout WITH (FORMAT csv, HEADER true)'.format(query=query)
+            result = recursive_read(self, copy_query)
+
+            query_columns = get_columns(self, query)
+            df_types = dtypes(query_columns, exclude_dates=True, exclude_the_geom=True)
+            date_column_names = date_columns_names(query_columns)
+
+            c = 0
+            for chunk in pd.read_csv(result, dtype=df_types,
+                                     chunksize=CHUNK_ROW_SIZE,
+                                     parse_dates=date_column_names,
+                                     true_values=['t'],
+                                     false_values=['f'],
+                                     index_col='cartodb_id' if 'cartodb_id' in df_types else False,
+                                     converters={'the_geom': lambda x: _decode_geom(x) if decode_geom else x}):
+                if c == 0:
+                    t.set_description('Downloading data from CARTO to a dataframe')
+                    df = chunk
+                    c += 1
+                else:
+                    df = df.append(chunk)
```

Comment:
pandas docs says that it's more efficient to do it like this:
```python
list_dfs = []
for i in range(10):
     ...
    list_dfs.append(df_chunk)

df_final = pd.concat(list_dfs)
```

See examples in https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.append.html

Reply:
Thanks! Didn't know about that 🙇